specification/iotcentral/data-plane/Microsoft.IoTCentral/stable/2022-05-31/iotcentral.json

```diff
@@ -0,0 +1,3968 @@
+            "description": "User patch body.",
+            "schema": {
+              "type": "object"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/User"
+            }
+          },
+          "default": {
+            "description": "An error response received from the IoT Central Service.",
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string",
+                "description": "Error code for specific error that occurred."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "delete": {
+        "operationId": "Users_Remove",
+        "summary": "Delete a user",
+        "x-ms-examples": {
+          "Delete a user": {
+            "$ref": "./examples/users_delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/Version"
+          },
+          {
+            "in": "path",
+            "name": "userId",
+            "description": "Unique ID of the user.",
+            "type": "string",
+            "required": true
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "Success"
+          },
+          "default": {
+            "description": "An error response received from the IoT Central Service.",
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string",
+                "description": "Error code for specific error that occurred."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "BaseDomain": {
+      "in": "path",
+      "name": "baseDomain",
+      "description": "The base domain for all Azure IoT Central service requests.",
+      "x-ms-parameter-location": "client",
+      "type": "string",
+      "x-ms-skip-url-encoding": true,
+      "default": "azureiotcentral.com",
+      "required": true
+    },
+    "CommandName": {
+      "in": "path",
+      "name": "commandName",
+      "description": "Name of this device command.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "ComponentName": {
+      "in": "path",
+      "name": "componentName",
+      "description": "Name of the device component.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "DeviceGroupId": {
+      "in": "path",
+      "name": "deviceGroupId",
+      "description": "Unique ID for the device group.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "maxLength": 255,
+      "pattern": "^[a-zA-Z0-9_-]*$",
+      "required": true
+    },
```

Comment:
Since it is used in PUT, do we have constraint other than the regex in patten?

From the pattern, it seems more about "device group name"?